### PR TITLE
Remove the hardcoded value for external distribution

### PR DESCRIPTION
**Pull request by berrygoudswaard**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description
The distribute_external flag was completely ignored because it was hardcoded set to 'true'. @negebauer already created an issue for this bug (see issue #8151).  

### Motivation and Context
**Why is this change required? What problem does it solve?**
This change makes it possible to distribute your app for internal testing on TestFlight. 

**If it fixes an open issue, please link to the issue here.**
See issue #8151 [https://github.com/fastlane/fastlane/issues/8151]

**Please describe in detail how you tested your changes.**
I have tested it by distributing an actual app for internal testing.

### CODE REVIEW

**Comment by KrauseFx** [comment]
I'd love to get a +1 on @tmm1 on this, as it was introduced with https://github.com/fastlane/fastlane/pull/5150 a while back 👍 

Thanks for your contribution @berrygoudswaard 🚀

**Review by tmm1 — APPROVED** [review]

**Review by negebauer — APPROVED** [review]

File: pilot/lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -61,7 +61,8 @@ def distribute(options, build = nil)
           return
         end
 
-        UI.message("Distributing build #{build.train_version}(#{build.build_version}) from #{build.testing_status} -> External")
+        type = options[:distribute_external] ? 'External' : 'Internal'
+        UI.message("Distributing build #{build.train_version}(#{build.build_version}) from #{build.testing_status} -> #{type}")
```
Comment (lacostej) [inline pilot/lib/pilot/build_manager.rb]: I wonder if we should also change the text below:

```ruby
       UI.message("Successfully distributed build to beta testers 🚀")
```
and replace "beta" with "External" or "Internal". While we are at it, maybe make it a `UI.success`

Comment (lacostej) [inline pilot/lib/pilot/build_manager.rb]: I also took note that spaceship doesn't seem to handle internal and external builds in the same way:
```ruby
      # @param (testing_type) internal or external
      def update_testing_status!(new_value, testing_type, build = nil)
        build ||= latest_build if testing_type == 'external'
```

But this isn't a problem here as the build is specified. Still weird though.

Comment (lacostej) [inline pilot/lib/pilot/build_manager.rb]: Will merge and do it in another PR

**Review by lacostej — APPROVED** [review]
